### PR TITLE
ndppd: 0.2.5 -> 0.2.5.43

### PR DESCRIPTION
**Pull request by Red54**
ndppd 0.2.5 is too old and lacks many features.
ndppd 0.2.5.43 means commit eb81b8f, 43 commits after 0.2.5.
https://github.com/DanielAdolfsson/ndppd/compare/0.2.5...eb81b8f

This is what SUSE has done for SLE (SUSE Linux Enterprise), and this version has been tested for many years, so stability should not be a problem.
https://packagehub.suse.com/packages/ndppd/
https://build.opensuse.org/package/show/network/ndppd
https://src.opensuse.org/pool/ndppd

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/services/networking/ndppd.nix [hunk 1]
```diff
@@ -91,7 +91,7 @@ let
           auto: Same as iface, but instead of manually specifying the outgoing
             interface, check for a matching route in /proc/net/ipv6_route.
         '';
-        default = "auto";
+        default = "static";
```
Comment (FliegendeWurst) [inline nixos/modules/services/networking/ndppd.nix:94]: Why change the default?

File: pkgs/applications/networking/ndppd/default.nix [hunk 2]
```diff
@@ -1,24 +1,21 @@
-{ lib, stdenv, fetchFromGitHub, gzip, nixosTests }:
+{ lib, stdenv, fetchurl, nixosTests }:
 
 stdenv.mkDerivation rec {
   pname = "ndppd";
-  version = "0.2.5";
+  version = "0.2.5.43";
```
Comment (FliegendeWurst) [inline pkgs/applications/networking/ndppd/default.nix:5]:
```suggestion
  version = "0.2.5-unstable-2023-12-11";
```
This is the nixpkgs convention for such versions.

File: pkgs/applications/networking/ndppd/default.nix [hunk 3]
```diff
@@ -1,24 +1,21 @@
...
 
-  src = fetchFromGitHub {
-    owner = "DanielAdolfsson";
-    repo = "ndppd";
-    rev = version;
-    sha256 = "0niri5q9qyyyw5lmjpxk19pv3v4srjvmvyd5k6ks99mvqczjx9c0";
+  src = fetchurl {
```
Comment (FliegendeWurst) [inline pkgs/applications/networking/ndppd/default.nix:7]: You can still use fetchFromGitHub with the commit revision.

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: nixos/modules/services/networking/ndppd.nix (hunk 1 above)
Comment (Red54) [reply nixos/modules/services/networking/ndppd.nix:94]: I mentioned it in the commit message, didn't you see it?
https://github.com/NixOS/nixpkgs/pull/352258/commits/47a329f255e60725bbd86451b9415e802e6114cd
https://manpages.opensuse.org/Tumbleweed/ndppd/ndppd.conf.5.en.html#static
Instead I'm asking why the default was set to "auto"?

File: pkgs/applications/networking/ndppd/default.nix (hunk 2 above)
Comment (Red54) [reply pkgs/applications/networking/ndppd/default.nix:5]: SUSE definition of 0.2.5.43 means commit eb81b8f, 43 commits after 0.2.5 (committed on 2017-09-20).
https://github.com/DanielAdolfsson/ndppd/compare/0.2.5...eb81b8f

File: pkgs/applications/networking/ndppd/default.nix (hunk 3 above)
Comment (Red54) [reply pkgs/applications/networking/ndppd/default.nix:7]: 如果我不使用SUSE的链接，世人怎知该版本是经过SUSE验证的版本？
If I don't use the SUSE link, how will the world know that this version is a SUSE-verified version?

File: pkgs/applications/networking/ndppd/default.nix (hunk 2 above)
Comment (Red54) [reply pkgs/applications/networking/ndppd/default.nix:5]: Has the 0.2.5-unstable-2023-12-11 you mentioned been verified? Can it be considered officially released?

Comment (FliegendeWurst) [reply pkgs/applications/networking/ndppd/default.nix:5]: I suppose not. In that case you can use 2017-09-20

File: pkgs/applications/networking/ndppd/default.nix (hunk 3 above)
Comment (FliegendeWurst) [reply pkgs/applications/networking/ndppd/default.nix:7]: You can mention that this is the SUSE packaged version in a comment.

File: nixos/modules/services/networking/ndppd.nix (hunk 1 above)
Comment (FliegendeWurst) [reply nixos/modules/services/networking/ndppd.nix:94]: Hmm. I don't know. That default was set in #55122.

**Comment by FliegendeWurst** [comment]
Ping for review: @fadenb

File: nixos/modules/services/networking/ndppd.nix (hunk 1 above)
Comment (Red54) [reply nixos/modules/services/networking/ndppd.nix:94]: Then set it to "static" to make things right.

File: pkgs/applications/networking/ndppd/default.nix (hunk 2 above)
Comment (Red54) [reply pkgs/applications/networking/ndppd/default.nix:5]: So you look at the version number 0.2.5-unstable-2017-09-20 and you can realize that this is the same version as SUSE's 0.2.5.43?
https://repology.org/project/ndppd/versions
https://pkgs.org/search/?q=ndppd

File: pkgs/applications/networking/ndppd/default.nix (hunk 3 above)
Comment (Red54) [reply pkgs/applications/networking/ndppd/default.nix:7]: As mentioned above, if the world can't realize that 0.2.5-unstable-2017-09-20 and 0.2.5.43 are the same version, then the SUSE link is what you need.

**Comment by Red54** [comment]
ndppd has tagged commit eb81b8f as 0.2.6
https://github.com/NixOS/nixpkgs/pull/355605

**Pull request closed without merging**